joss-paper/paper.md

```diff
@@ -0,0 +1,130 @@
+---
+title: 'impedance.py: A Python package for electrochemical impedance analysis'
+tags:
+  - Python
+  - electrochemistry
+  - impedance
+  - lithium-ion batteries
+  - fuel cells
+  - corrosion
+authors:
+  - name: Matthew D. Murbach
+    email: matt@hivebattery.com
+    orcid: 0000-0002-6583-5995
+    affiliation: 1
+  - name: Brian Gerwe
+    email: brian.s.gerwe@gmail.com
+    orcid: 0000-0002-1184-8483
+    affiliation: 2
+  - name: Neal Dawson-Elli
+    affiliation: 3
+  - name: Lok-kun Tsui
+    email: lktsui@unm.edu
+    orcid: 0000-0001-7381-0686
+    affiliation: 4
+affiliations:
+ - name: Hive Battery Labs
+   index: 1
+ - name: University of Washington, Department of Chemical Engineering
+   index: 2
+ - name: PayScale, Inc.
+   index: 3
+ - name: University of New Mexico, Center for MicroEngineered Materials
+   index: 4
+date: 14 March 2020
+bibliography: paper.bib
+---
+
+`impedance.py` is a community-driven Python package for making the analysis of electrochemical impedance spectroscopy (EIS)  data easier and more reproducible. `impedance.py` currently provides several useful features commonly used in the typical  impedance analysis workflow:
+
+- _data preprocessing_: functions for importing data from a variety of instruments and file types
+- _data validation_: easy-to-use methods for checking measurement validity
+- _model fitting_: a simple and powerful interface for quickly fitting models to analyze data
+- _model selection_: parameter error estimates and model confidence bounds
+- _visualization_: interactive and publication-ready Nyquist and Bode plots
+
+# Background
+
+Electrochemical impedance spectroscopy (EIS) is a powerful technique for noninvasively probing the physicochemical processes governing complex electrochemical systems by measuring the relationship between voltage and current (i.e. the impedance) as a function of frequency [@orazem_electrochemical_2008]. Analysis of EIS spectra often involves the construction and fitting of models to represent the system response. Equivalent circuit models, a popular method for interpreting impedance spectra, represent processes in an electrochemical system with elements in a circuit allowing quantitative characterization of the data. EIS inherently requires the system response is linear, causal and stable over the measurement time. As such, researchers must also ensure their data complies with these assumptions – often through some implementation of Kramers-Kronig analysis. Following the approach outlined above, workers have applied EIS to problems ranging from  measuring the dielectric properties of biological systems to corrosion rates of coated metals. More recently, EIS has been widely adopted for studying processes governing batteries and fuel cells.
```

Comment:
We should add references here for the stated applications.

Can't find the specific paper for dielectric properties of biological systems. Anyone know which paper this is talking about?

Corrosion: https://www.sciencedirect.com/science/article/pii/0013468695004408 

Batteries: https://www.sciencedirect.com/science/article/abs/pii/S0378775399003511 + https://www.sciencedirect.com/science/article/pii/S0013468605007899

Fuel Cells: https://iopscience.iop.org/article/10.1149/1.1836485/meta + https://www.sciencedirect.com/science/article/pii/S0013468698001388

Reply:
Thanks for those references! I'll add them to my local version now. The measurement of dielectric properties of biological systems was mentioned in the intro of Mark Orazem's EIS book, so I'll try to find the reference used in that book.

Reply:
These references were added in [this commit](https://github.com/ECSHackWeek/impedance.py/pull/91/commits/78fd7c5c291dba03491e0dc59b898838539d4c28).